### PR TITLE
fix: internal indices for classes `_Idempotency` and `_Role` are not protected in defined schema

### PR DESCRIPTION
**Pull request by Moumouls**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Missing protected indexes detection on _Role and _Idempotency class

Related issue: NONE

### Approach

Fix and cover with unit tests

### TODOs before merging

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

File: spec/DefinedSchemas.spec.js [hunk 1]
```diff
@@ -432,6 +432,40 @@ describe('DefinedSchemas', () => {
...
+
+    it('should detect protected indexes for _User class', () => {
+      const definedSchema = new DefinedSchemas({}, {});
+      const protectedUserIndexes = ['_id_', 'case_insensitive_email', 'username_1', 'email_1'];
+      protectedUserIndexes.forEach(field => {
+        expect(definedSchema.isProtectedIndex('_User', field)).toEqual(true);
+      });
+      expect(definedSchema.isProtectedIndex('_User', 'test')).toEqual(false);
+    });
+    it('should detect protected indexes for _Role class', () => {
```
Comment (mtrezza) [inline spec/DefinedSchemas.spec.js]:
```suggestion
    });

    it('should detect protected indexes for _Role class', () => {
```

File: spec/DefinedSchemas.spec.js [hunk 2]
```diff
@@ -432,6 +432,40 @@ describe('DefinedSchemas', () => {
...
+      const definedSchema = new DefinedSchemas({}, {});
+      expect(definedSchema.isProtectedIndex('_Role', 'name_1')).toEqual(true);
+      expect(definedSchema.isProtectedIndex('_Role', 'test')).toEqual(false);
+    });
+    it('should detect protected indexes for _Idempotency class', () => {
```
Comment (mtrezza) [inline spec/DefinedSchemas.spec.js]:
```suggestion
    });

    it('should detect protected indexes for _Idempotency class', () => {
```

**Review by mtrezza — DISMISSED** [review]
Looks good! I changed the PR title to be more descriptive as a changelog entry - does it make sense?

**Comment by Moumouls** [comment]
> does it make sense?

Yes it's okay @mtrezza 

Thanks for your commit suggestions

**Review by mtrezza — APPROVED** [review]
Looks good! Thanks for the fix! Re-running CI until all tests pass...

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.3.0-alpha.21](https://github.com/parse-community/parse-server/releases/tag/5.3.0-alpha.21)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/5.4.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.4.0](https://github.com/parse-community/parse-server/releases/tag/5.4.0)

**Pull request merged**